### ISSUE TITLE
About the voting accuracy?

### ISSUE THREAD

**Issue by liruihui** [Yochengliu/Relation-Shape-CNN#13]
I have got the best validation accuracy with 92.1777%. But when I test the model with voting with random scaling as paper, the accuracy drops to 91.1%.  

Could you share your voting code? @Yochengliu  

Thanks.

**Comment by Hlxwk** [comment]
I meet the same issue, and could I see your test code in ModelNet40? @liruihui @Yochengliu

**Comment by liruihui** [comment]
![image](https://user-images.githubusercontent.com/31917103/66319336-87b53a00-e94f-11e9-8a58-f2349ebfcb70.png)
This is my test code based the paper.

**Comment by Hlxwk** [comment]
@liruihui 
'scale' in Table 4 of @Yochengliu 's paper should refer to multi-scale grouping (different from MSG in PointNet++), not data processing. Thus you should retrain the ModelNet40 codes with multi-scale grouping, noticing the performance achieve 92.9% or not. Then you could test the results with ten votes.

![2019-10-07 22-35-11屏幕截图](https://user-images.githubusercontent.com/28684150/66321209-cef0fa00-e952-11e9-8dac-6554ac893679.png)

**Comment by liruihui** [comment]
![image](https://user-images.githubusercontent.com/31917103/66360839-14431500-e9af-11e9-95e9-18f21f3d3f4c.png)
They conduct the testing with random scaling and average the predictions. I just use the release code with a single scale. But after the voting, the performance drops. It should not be that case.

**Comment by Hlxwk** [comment]
@liruihui 
I do know that, but I think the model with MSG training would be better than your model without MSG training when runing ten votes with random scaling.

**Comment by liruihui** [comment]
@Hlxwk 
I do not think the MSG is so decisive. Even you replace the RS-CNN with raw shared-mlps, it also works very well.

**Comment by Hlxwk** [comment]
@liruihui 
Maybe, you are right. I am also curious about ten votes, but I met the same accuracy drop when runing ten-votes tests. Hope @Yochengliu  could share his ten-votes codes.

**Comment by ShiQiu0419** [comment]
For the released codes, I also got 92.2%. 

For the scale part, it seems like extracting rs features from different scales of neighborhood. 

For the ten votes, I also think it is not reliable. Acturally the codes for multi voting have been implemented in both PointNet and PointNet++, and I tried them. According to my experience, sometimes ten voting would boost your accuracy a little bit, while sometimes would lower down the accuracy. It depends on the "robustness" of your trained model.

**Comment by Yochengliu** [comment]
Hi, all, we have uploaded the voting script, please have a check and try it. Hope it helpful.

**Comment by liruihui** [comment]
Thanks for your sharing. @Yochengliu 
Could you release your multi-scale classification code? Because I can not find any config details about that.

**Comment by sheshap** [comment]
@liruihui can you please release the multi-scale classification code if you coded it yourself using the configurations provided in the arxiv version of RS-CNN ?
